types/src/v0/impls/block/full_payload/payload.rs

```diff
@@ -242,21 +238,7 @@ impl QueryablePayload<SeqTypes> for Payload {
         vid: &VidCommonQueryData<SeqTypes>,
         index: &Index,
     ) -> Option<Self::InclusionProof> {
-        let common = match vid.common() {
-            VidCommon::V0(common) => common,
-            VidCommon::V1(_) => {
-                // TODO HACK! THE RETURNED PROOF MIGHT FAIL VERIFICATION.
-                // https://github.com/EspressoSystems/hotshot-query-service/issues/639
-                //
-                // Need a `ADVZCommon` to proceed. Need to modify return type to accept either
-                // version of VID common and produce a proof from the corresponding version.
-                &hotshot_types::vid::advz::advz_scheme(10)
-                    .disperse(&self.raw_payload)
-                    .unwrap()
-                    .common
-            },
-        };
-        let proof = TxProof::new(index, self, common)?.1;
+        let proof = TxProof::new(index, self, vid.common())?.1;
```

Comment:
what about we do the version dispatch centrally here, so that `tx_proof/avidm.rs`'s method will directly takes in `common: avidm::AvidMCommon` and don't have to add separate version detection both in advz and avidm.